doc/contributing/hw/comportability/README.md

```diff
@@ -462,62 +462,137 @@ Here is an example specification:
+This is distinct from another possible model in which each peripheral would send only one interrupt, and the processor would disambiguate by querying the peripheral to figure out which interrupt was triggered.
+
+### Register Creation
+
+By default for every peripheral, three registers are **automatically** created to manage the interrupts (as defined in the `interrupt_list` portion of the Hjson file).
+These registers are named `INTR_STATE`, `INTR_ENABLE` and `INTR_TEST`, and are placed at the top of the peripheral's address map in that order automatically by the `reggen` tool.
+Every interrupt has one bitfield in each of three registers.
+(It is currently an error if the `.hjson` defines more than 32 interrupts per-peripheral.)
+
+| Name          | Offset | Desc                                                                                  |
+|---------------|--------|---------------------------------------------------------------------------------------|
+| `INTR_STATE`  | 0x0    | holds the current state of the interrupt (may be RO or W1C depending on "IntrT").     |
+| `INTR_ENABLE` | 0x4    | enables/masks the output of INTR_STATE to create the wired interrupt signal `intr_o`. |
+| `INTR_TEST`   | 0x8    | write-only (`wo`) register which asserts the interrupt for testing purposes.          |
+
+The contents of the `INTR_STATE` register are not qualified by `INTR_ENABLE`, but rather show the raw state of all latched hardware interrupt events.
+
+### CIP Interrupt Types
+
+Interrupts are a mechanism for hardware to request attention from software, commonly as either as a normal part of a system's operation or as a way to handle exceptional circumstances.
+To design robust control flows and abstractions using interrupts, one way is to model Interrupts + Interrupt Handlers as forming a handshaking interaction between hardware and software.
+Hardware creates an interrupt request when a condition is true or an event occurs, and software responds to this request, possibly performing some action to address the condition/event, then acknowledges the interrupt request as completed.
+This request/acknowledge sequence is one-way to mitigate the effects of non-determinism in a real system, such as race-conditions due to the inherent round-trip latency from signalling to handling, and the unknown latency of a core executing the handler for one of possibly many competing interrupt requests.
+
+When creating an interrupt in a peripheral, we can identify two distinct behaviours a designer may choose to implement.
+
+1. **Event** type: Whenever an instantaneous event occurs trigger a new interrupt request, but do not create a new request until the previous request has been acknowledged (ack is a write to `INTR_STATE`, which is `W1C` in this mode).
+2. **Status** type: So long as a condition is persistently true, trigger a new interrupt request, including immediately after the previous request has been acknowledged.
```

Comment:
I'm not sure about "new" here. Isn't this just "Trigger the interrupt request when a condition is true"?